R/promptAndRun.R

```diff
@@ -9,23 +9,29 @@ promptAndRun <- function(mydir = ".", user = NULL, daysback = 3) {
     loopRuns(choose_folder("."), user = user)
   } else if (mydir == "-f") {
     loopRuns(dir(), user = user)
-  } else if (mydir == "-cr") {
+  } else if (mydir %in% c("-cr", "-ar")) {
     myruns <- system(paste0("squeue -u ", user, " -h -o '%Z'"), intern = TRUE)
     runnames <- system(paste0("squeue -u ", user, " -h -o '%j'"), intern = TRUE)
 
-    myruns2 <- system(paste0("sacct -u ", user, " -s cd,f,cancelled,timeout,oom -S ", as.Date(format(Sys.Date(), "%Y-%m-%d")) - as.numeric(daysback), " --format WorkDir -P -n"), intern = TRUE)
- #   myruns2<-myruns2[!grepl("^$",myruns2)]
-    myruns <- c(myruns, myruns2)
-    runnames2 <- system(paste0("sacct -u ", user, " -s cd,f,cancelled,timeout,oom -S ", as.Date(format(Sys.Date(), "%Y-%m-%d")) - as.numeric(daysback), " --format JobName -P -n"), intern = TRUE)
-#    runnames2<-runnames2[!grepl("^batch$",runnames2)]
-    runnames <- c(runnames, runnames2)
+    if (mydir == "-cr") {
+      myruns2 <- system(paste0("sacct -u ", user, " -s cd,f,cancelled,timeout,oom -S ", as.Date(format(Sys.Date(), "%Y-%m-%d")) - as.numeric(daysback), " --format WorkDir -P -n"), intern = TRUE)
+      # myruns2<-myruns2[!grepl("^$",myruns2)]
+      myruns <- c(myruns, myruns2)
+      runnames2 <- system(paste0("sacct -u ", user, " -s cd,f,cancelled,timeout,oom -S ", as.Date(format(Sys.Date(), "%Y-%m-%d")) - as.numeric(daysback), " --format JobName -P -n"), intern = TRUE)
+      # runnames2<-runnames2[!grepl("^batch$",runnames2)]
+      runnames <- c(runnames, runnames2)
+    }
+
     if (any(grepl("mag-run", runnames))) {
```

Comment:
here, I had to change the logic a bit, because if no runname is "batch", as `which(1 %in% c(2)) = integer(0)` and `myruns[-integer(0)]` is always an empty vector (which is strange!)

Reply:
ok